### PR TITLE
move collective tests into a collective directory

### PR DESCRIPTION
**Pull request by sljlp**
### PR types

Others
### PR changes

Others
### Describe

Move all collective unit tests into a collective directory.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/CMakeLists.txt [hunk 1]
```diff
@@ -1228,7 +1190,7 @@ endif()
 if(WITH_XPU_BKCL)
   #py_test(test_collective_reduce_api_xpu SRCS "test_collective_reduce_api.py")
   py_test(test_collective_allreduce_api_xpu
-          SRCS "test_collective_allreduce_api.py")
+          SRCS "collective/test_collective_allreduce_api.py")
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/CMakeLists.txt:960]: Should it be moved?

File: python/paddle/fluid/tests/unittests/CMakeLists.txt [hunk 2]
```diff
@@ -1646,35 +1591,11 @@ if((WITH_ROCM OR WITH_GPU) AND NOT WIN32)
...
-      test_collective_allreduce_api
-      test_new_group_api
-      test_collective_broadcast_api
-      test_collective_allgather_api
-      test_collective_allgather_object_api
-      test_collective_alltoall_api
-      test_collective_global_gather
-      test_collective_global_scatter
-      PROPERTIES LABELS "RUN_TYPE=DIST")
+    set_tests_properties(test_pipeline_parallel PROPERTIES LABELS
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/CMakeLists.txt:1302]: Should it be moved?

**Review by gongweibao — CHANGES_REQUESTED** [review]
Rename not copy.

**Review by XieYunshen — DISMISSED** [review]
LGTM 单测迁移

**Review by gongweibao — CHANGES_REQUESTED** [review]
1. This tool should manage `dist_port` and the user need only to consider the ports they would use.
2. Remove the commented lines in the CMakeLists.txt
3. Add a key `device_num`?
4. Add a sh script under each dir which needs it?

**Review by XieYunshen — DISMISSED** [review]
LGTM

**Review by gongweibao — DISMISSED** [review]
LGTM

**Review by wanghuancoder — DISMISSED** [review]
LGTM

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]

**Review by wanghuancoder — APPROVED** [review]
LGTM

**Review by gongweibao — APPROVED** [review]
LGTM

**Pull request merged**